girder/models/file.py

```diff
@@ -161,6 +161,13 @@ def validate(self, doc):
                 raise ValidationException(
                     'Linked file URL must start with http: or https:.',
                     'linkUrl')
+        if doc.get('assetstoreModel'):
+            # If assetstore model is overridden, make sure it's a valid model
+            try:
+                self.model(doc['assetstoreModel'], plugin=doc.get('assetstoreModelPlugin'))
```

Comment:
Elsewhere we have `attachedToType` where the type is either a string with the name of the model OR a list that has two entries that can be passed to `self.model(*file['attachedToType'])`.  In the utilities search module, we also handle model types specified by a dot-separated string.  We should pick a method to be the standard (and possible modify the `ModelImporter.model` method to handle all of these directly so we can always just call `ModelImported.model(<string or dot string or list or tuple with one or two items>)`).

Reply:
Good point... what do you think we should do specifically for this PR?

Reply:
I think, to match elsewhere in the code, this should be a single parameter called `assetstoreType` (rather than `assetstoreModel` and `assetstoreModelPlugin`).  Which means that you'd have to handle the string or tuple cases here (unless you want to make `model()` handle them generically, but I figured that would be better as a separate PR).

Reply:
So if it's a plain string, it would be a core model, but if it's a tuple we assume it's of the form `(modelName, pluginName)`?

Reply:
That's what we've done with `attachedToType`.